### PR TITLE
Changed environment-variable.md to add julia_num_threads description and edits to object.md to add a warning of outdated info

### PR DESCRIPTION
**Pull request by TakshDhabalia**
Added some lines about Julia_Num_Threads in Environmental variables as per the open issue [48960](https://github.com/JuliaLang/julia/issues/48960) ,not added the other suggestion . open to suggestions .

### CODE REVIEW

**Comment by TakshDhabalia** [comment]
also added a small change to the docs of object.md to put out of date till we update the documentation of the allocobj and other out of date stuff as per issue [#48784](https://github.com/JuliaLang/julia/issues/48784) and the comments of the same

**Comment by Tokazama** [comment]
This helps people not go down a rabbit whole trying to figure out this methods, but it really just makes it seem like this entire section is useless now.

As long as the relevant methods for these aren't changing drastically anytime soon we should probably just have the actual method names updated with a warning that they are internal and an unstable API.

**Comment by TakshDhabalia** [comment]
Hey, thanks for the feedback @Tokazama . Could you suggest what I do for this? Just update the out of date mentions Or something else?

**Comment by Tokazama** [comment]
It's a bit tricky figuring out what to do here because even the methods themselves aren't very well documented. You can sort of go off of what methods were important enough to include in "jl_exported_funcs.inc" but I'm still not sure. @MichaelHatherly appears to be the last person in the git blame on that line (7 years ago). He may have some feedback.

**Comment by MichaelHatherly** [comment]
> @MichaelHatherly appears to be the last person in the git blame on that line (7 years ago). He may have some feedback.

Those changes would have been from https://github.com/JuliaLang/julia/pull/18588 which was just part of the mass conversion from rst to md for the docs. So content-wise for this particular change I'd not have much to say on them.

File: doc/src/devdocs/object.md [hunk 1]
```diff
@@ -185,6 +185,9 @@ jl_value_t *newobj(jl_value_t *type, size_t nfields);
 And at the lowest level, memory is getting allocated by a call to the garbage collector (in `gc.c`),
 then tagged with its type:
 
+!!! note "Out of date Warning"
+    The documentation and usage of `jl_gc_allocobj` may be out of date
+
```
Comment (inkydragon) [inline doc/src/devdocs/object.md:190]: Perhaps it would be better to split into separate pr.

**Comment by inkydragon** [comment]
> You can sort of go off of what methods were important enough to include in "jl_exported_funcs.inc" but I'm still not sure.

In fact, `jl_gc_allocobj` is a function that has been exported but rarely used, and `jl_gc_alloc` has not been exported but is heavily used.

So `devdoc` and some internal api are already a rabbit hole.

File: doc/src/devdocs/object.md (hunk 1 above)
Comment (TakshDhabalia) [reply doc/src/devdocs/object.md:190]: sure, that would be better then merging in one pr to avoid confusion

**Comment by Tokazama** [comment]
I agree that the environment variables portion should be put in a separate PR. I think there will need to be several steps to resolve updating docs related to all C code.

I think the first step is updating contributor guidelines for C code. Even something as general as "See Base.ccall_wrapper_function or details" so that there's some relevant documentation for methods that are critical parts of any API.
